src/test/java/com/exadel/demo/controller/UserControllerTest.java

```diff
@@ -0,0 +1,182 @@
+        MockMvcBuilders.standaloneSetup(this.userController)
+                .build()
+                .perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().contentType("application/json"))
+                .andExpect(MockMvcResultMatchers.content().string("[]"));
+    }
+
+    @Test
+    void testGetAllUsers2() throws Exception {
+        doNothing().when(this.logger).info((String) any());
+        when(this.userService.getAllUsers()).thenReturn(new ArrayList<>());
+        MockHttpServletRequestBuilder getResult = MockMvcRequestBuilders.get("/user/getAll");
+        MockMvcBuilders.standaloneSetup(this.userController)
+                .build()
+                .perform(getResult)
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().contentType("application/json"))
+                .andExpect(MockMvcResultMatchers.content().string("[]"));
+    }
+
+    @Test
+    void testGetUserById() throws Exception {
+        doNothing().when(this.logger).info((String) any());
+        when(this.userService.getUser((Long) any())).thenReturn(new UserDto());
+        MockHttpServletRequestBuilder getResult = MockMvcRequestBuilders.get("/user/get/*");
+        MockHttpServletRequestBuilder requestBuilder = getResult.param("id", String.valueOf(1L));
+        MockMvcBuilders.standaloneSetup(this.userController)
+                .build()
+                .perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().contentType("application/json"))
+                .andExpect(MockMvcResultMatchers.content().string("{\"firstName\":null,\"lastName\":null,\"email\":null,\"roleDto\":null}"));
+    }
+
+    @Test
+    void testGetUserById2() throws Exception {
+        doNothing().when(this.logger).info((String) any());
+        when(this.userService.getUser((Long) any())).thenReturn(new UserDto());
+        MockHttpServletRequestBuilder getResult = MockMvcRequestBuilders.get("/user/get/*");
+        MockHttpServletRequestBuilder requestBuilder = getResult.param("id", String.valueOf(1L));
+        MockMvcBuilders.standaloneSetup(this.userController)
+                .build()
+                .perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().contentType("application/json"))
+                .andExpect(MockMvcResultMatchers.content().string("{\"firstName\":null,\"lastName\":null,\"email\":null,\"roleDto\":null}"));
+    }
+
+    @Test
+    void testSaveUser() throws Exception {
+        doNothing().when(this.logger).info((String) any());
+        when(this.userService.addUser((UserDto) any())).thenReturn(new UserDto());
+
+        UserDto userDto = new UserDto();
```

Comment:
below you have  something similar in test testUpdateUser it better to create it on top somewhere in private method like builder, or maybe mock this object and use it